### DISCUSSION TITLE
Standardizing Artisan Command and Scheduler Argument Syntax

### DISCUSSION THREAD

**Discussion by hamidrezaniazi** [laravel/framework#49643] [Ideas]
## Current State
Consider an Artisan command with the signature:

```PHP
protected $signature = 'command {foo} {--bar=}';
```

When called directly using the `InteractsWithConsole` trait, the syntax is as follows:

```PHP
$this->artisan(Command::class, [
    'foo' => 'bleep',
    '--bar' => 'bloop',
]);
```

However, when scheduling the command with the Laravel Scheduler, the syntax becomes:

```PHP
$schedule->command(Command::class, [
    'bleep',
    '--bar' => 'bloop',
]);
```

## Proposed Solution
To enhance consistency and reduce the chances of misuse, I propose standardizing the syntax for specifying command arguments and options, regardless of whether the command is called directly or scheduled.

This change would result in a unified syntax, such as:

```PHP
$this->artisan(Command::class, [
    'foo' => 'bleep',
    '--bar' => 'bloop',
]);

// AND

$schedule->command(Command::class, [
    'foo' => 'bleep',
    '--bar' => 'bloop',
]);
```
## Discussion
Please share your thoughts on this proposed change. Do you see any potential drawbacks or advantages?

**Comment by Rikcon** [comment]
Hello, encountered this inconsistency today, really really frustrating, had to dig out night logs to see why it didn't work, while looking through logs my eye stumbled upon full command that it calls using $schedule->command(
Thank you for reporting this so i don't have to !

**Comment by crynobone** [comment]
`artisan` method utilises `symfony/console` and have full access to Command arguments and options definition while `$schedule` only utilises `symfony/process` and is bound to it's limitation.

**Reply by Rikcon** [reply]
Understood, maybe some warning about this in this section of the docs [](https://laravel.com/docs/11.x/scheduling#scheduling-artisan-commands) would be sufficient ? 
I'm happy to provide a PR but wordsmithing isn't one of my strong skills )
 
I see that in an example `Schedule::command(SendEmailsCommand::class, ['Taylor', '--force'])->daily();` there is a hint that this is the right way to pass parameters in scheduled command, but i think additional sentence about this inconsistency would be good.

**Comment by willrowe** [comment]
This has happened to me as well and is very frustrating until you figure it out.
